vortex-array/src/array/varbin/flatten.rs

```diff
@@ -1,10 +1,94 @@
+use std::sync::Arc;
+
+use arrow_array::builder::GenericByteViewBuilder;
+use arrow_array::types::BinaryViewType;
+use arrow_array::{Array, ArrayRef};
 use vortex_error::VortexResult;
 
 use crate::array::varbin::VarBinArray;
-use crate::{Canonical, IntoCanonical};
+use crate::array::varbinview::{BinaryView, VarBinViewArray};
+use crate::arrow::FromArrowArray;
+use crate::validity::ArrayValidity;
+use crate::{ArrayData, Canonical, IntoArray, IntoArrayData, IntoCanonical};
 
 impl IntoCanonical for VarBinArray {
     fn into_canonical(self) -> VortexResult<Canonical> {
-        Ok(Canonical::VarBin(self))
+        fn into_byteview(array: &VarBinArray) -> ArrayRef {
+            let mut builder = GenericByteViewBuilder::<BinaryViewType>::with_capacity(array.len());
```

Comment:
this is wrong: this is only if the max(offsets) of the varbin is <= u32::MAX

if it's >u32::MAX (i.e. if more than 4GB of strings in a single array) then should construct via the iterator